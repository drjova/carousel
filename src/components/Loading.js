import React from 'react';
import PropTypes from 'prop-types';

const Loading = ({ loading }) => loading && <div>Loading</div>;

Loading.propTypes = {
  loading: PropTypes.bool.isRequired,
};

export default Loading;
