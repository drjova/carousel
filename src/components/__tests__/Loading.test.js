import React from 'react';
import { mount } from 'enzyme';

import Loading from './../Loading';

describe('Loading component', () => {
  it('should render ``Loading``', () => {
    const component = mount(<Loading loading />);
    expect(component).toMatchSnapshot();
  });

  it('should render nothing', () => {
    const component = mount(<Loading loading={false} />);
    expect(component).toMatchSnapshot();
  });
});
