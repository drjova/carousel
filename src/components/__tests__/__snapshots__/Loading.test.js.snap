// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render \`\`Loading\`\` 1`] = `
<div>
  Loading
</div>
`;

exports[`Loading component should render nothing 1`] = `""`;
